tests/lib/muinstaller: add passphrase authentication support

ZeyadYasser opened the pull request:
This PR adds passphrase authentication support to the muinstaller.

~~**NOTE: Currently go.mod is updated to get the snapd libraries from my branch for the client library bits, I will adjust it to point the fde branch when https://github.com/canonical/snapd/pull/14845 is merged.**~~

For reference check the API spec: SD201

andrewphelpsj reviewed (approved):
Thank you! Make sure to check on the relevant nested tests before merging.

ZeyadYasser commented:
It seems that the failing muinstaller test is because muinstaller is built on the test machine and not through snapcraft, so it cannot point to my branch with API support. I will re-run the tests when https://github.com/canonical/snapd/pull/14845 is merged.

bboozzoo reviewed (approved):
LGTM

bboozzoo commented:
labeled as blocked until https://github.com/canonical/snapd/pull/14845 lands

ZeyadYasser commented:
PR is now rebased on top of https://github.com/canonical/snapd/pull/14845, and muinstaller go.mod points snapd library from the fde branch.

Removed the blocked label.
